backends/build_system/exe.py

```diff
@@ -0,0 +1,80 @@
+import os
+import subprocess
+from typing import Callable, List, Dict, Any
+from dataclasses import dataclass, field
+
+from constants import EXE_ASSETS_DIR, PYINSTALLER_DIR, DISTRIBUTION_SOURCE
+from utils import Utils
+from awscli_venv import AwsCliVenv
+
+
+@dataclass
+class ExeBuilder:
+    workspace: str
+    venv: AwsCliVenv
+
+    _exe_dir: str = field(init=False)
+    _final_dist_dir: str = field(init=False)
+    _dist_dir: str = field(init=False)
+    _build_dir: str = field(init=False)
+
+    _utils: Utils = field(default_factory=lambda: Utils())
+
+    def __post_init__(self):
+        self._exe_dir = os.path.join(self.workspace, "aws")
+        self._final_dist_dir = os.path.join(self._exe_dir, "dist")
+        self._dist_dir = os.path.join(self.workspace, "dist")
+        self._build_dir = os.path.join(self.workspace, "build")
+
+    def build(self, cleanup=True):
+        self._ensure_no_existing_build_dir()
+        self._build_aws()
+        self._build_aws_completer()
+        self._utils.copy_directory_contents_into(EXE_ASSETS_DIR, self._exe_dir)
+        if cleanup:
+            self._cleanup()
+        print(f"Built exe at {self._exe_dir}")
+
+    def _ensure_no_existing_build_dir(self):
+        if self._utils.isdir(self._dist_dir):
+            self._utils.rmtree(self._dist_dir)
+
+    def _build_aws(self):
+        aws_exe_build_dir = self._run_pyinstaller("aws.spec")
+        self._utils.update_metadata(
+            aws_exe_build_dir,
+            distribution_source=DISTRIBUTION_SOURCE,
```

Comment:
Not needed for this PR, but something we should consider... For installations from the source distribution, we should probably update the distribution metadata to be unique for both `portable-exe` and `system-sandbox` source installs. Right now it seems if you choose `portable-exe`, the metadata will be `exe`, and if you choose `system-sandbox`, it will be `source`. However, this makes it difficult to differentiate between whether users are using the source dist or our public exes (for `portable-exe`) or direct `pip install` (for `system-sandbox`)